feat(IT-41): Translation Language Selection

qiningliu1 opened the pull request:
###  Features Implemented:
- Translation language selector including 10 languages 
- Dynamic translation updating and realtime switching 
- Use translation language hook for language preference 
- Local storage for language preference

 Add useEffect dependency to ensure real-time language switching

### UI & Layout:
- Implemented the following:

  - **Language** (dropdown/select input)
  - **Confirmed button** 
 

![language-selector](https://github.com/user-attachments/assets/8b13eb74-3e73-4916-831b-87b1b4641396)


### Dynamic Translation Update
-  Real-time language switching during meetings
-  All subsequent translations use newly selected language
-  Chat bubbles continue showing original + translated text
-  Fixed useEffect dependency issue for immediate language updates
![language-testing](https://github.com/user-attachments/assets/f1af5f8f-f070-4775-bab9-d84559dc8912)

![hands](https://github.com/user-attachments/assets/6a9c8960-fdce-4776-9a9d-dcf38aee7aeb)



### Key Technical Changes:
1. **Custom Hook**: `useTranslationLanguage` for state management and local storage
2. **UI Component**: Bottom sheet language selector
3. **Bug Fix**: Added `localPreferenceLanguage` to useEffect dependencies
4. **Integration**: Connected language selector to AI translation button


### Code Review change
- Remove all console.log statements
- Clean up HTML comments and empty lines
- Improve accessibility implementation:
  * Enhance NavIconButton component interface to accept `alt` parameter
  * Add alt text for all navigation icons in bottom-nav-bar.tsx
- Use useCallback for performance optimization in speech recognition handlers
- Remove backend dependency for language preference sync (pure frontend solution)
- Fix ESLint accessibility warnings with proper alt attributes






Related task: IT-41

StevenRen929 commented:
Except write down the missing dependencies in useEffect.  Please write down the reason to use useCallback hook in recognising and recognised text. And also write down the multiple users join room status and speaking status testing results picture to ensure the dev branch meeting room function works fine in this pull request. Thanks

qiningliu1 committed:
feat(IT-41):Complete-Translation-Language-Selector
- Add translation language selector supporting 10 languages
- Implement dynamic translation switching with real-time updates
- Add history message retranslation functionality
- Integrate with useTranslationLanguage hook for language preference management
- Support localStorage persistence
- Add responsive bottom sheet UI

- Merge latest dev branch features
- Fix accessibility issues

- Remove all console.log statements
- Clean up HTML comments and empty lines
- Improve accessibility implementation:
  * Enhance NavIconButton component interface to accept `alt` parameter
  * Add alt text for all navigation icons in bottom-nav-bar.tsx
- Use useCallback for performance optimization in speech recognition handlers
- Remove backend dependency for language preference sync (pure frontend solution)
- Fix ESLint accessibility warnings with proper alt attributes

- feat: logout button in landing page - mobile and PC
- feat: Quit meeting to Dashboard page and Logout

Chris-Wang commented on src/app/module/instant-meeting-room/meeting-room.tsx:
```diff
@@ -29,21 +32,81 @@ interface userStatus {
+        return;
+      }
+
+      console.log(
```
remove console.log

Chris-Wang commented on src/app/module/instant-meeting-room/main-translation-area/main-translation-area.tsx:
```diff
@@ -68,14 +80,36 @@ function MainTranslationArea(speechData: speechData) {
-            afterTranslation={translatedMessage[index]}
-          />
-        ))}
+        {/*Retranslation loading indicator */}
```
As we discussed long time ago, we don't want comments and emply lines in html section

Chris-Wang commented on src/app/module/instant-meeting-room/meeting-room.tsx:
```diff
@@ -235,6 +299,30 @@ function MeetingRoom() {
+
+  const handleLanguageSelect = (languageCode: string) => {
+    updateLanguage(languageCode);
+    console.log('Translation language updated to:', languageCode);
```
remove console.log

Chris-Wang commented on src/app/module/instant-meeting-room/translation-language-selector.tsx:
As we discussed long time ago, we don't want comments and emply lines in html section

Chris-Wang commented on src/app/module/common/nav-icon-button.tsx:
```diff
@@ -19,7 +19,7 @@ function NavIconButton({ src, bg, onClick }: NavIconButtonProps) {
       onClick={onClick}
     >
-      <Image w="60%" h="60%" objectFit="contain" src={src} />
+      <Image w="60%" h="60%" objectFit="contain" src={src} alt="" />
```
why add this?

Chris-Wang commented on src/app/module/instant-meeting-room/meeting-room.tsx:
```diff
@@ -29,21 +32,81 @@ interface userStatus {
+          if (index === recognizedText.length - 1) {
+            setTimeout(() => {
+              setIsRetranslating(false);
+              console.log('History message retranslation completed');
```
remove console.log

qiningliu1 committed:
fix: add missing alt attribute to share button - Resolves TypeScript error about required prop
